### PR TITLE
blog: Add Implementing the Idempotency-Key specification on Apache AP…

### PR DESCRIPTION
**Pull request by nfrankel**
Screenshot 2024-04-09 at 16 01 04

### CODE REVIEW

**Review by moonming — DISMISSED** [review]

File: blog/en/blog/2024/04/11/implement-idempotency-key-apisix.md [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+  - IETF
+  - specification
+  - plugin
+  - coding
+description: >
+  Last week, I wrote an analysis of the IETF Idempotency-Key specification. The specification aims to avoid duplicated requests. In short, the idea is for the client to send a unique key along with the request:
+  If the server doesn't know the key, it proceeds as usual and then stores the response.
+  If the server knows the key, it short-circuits any further processing and immediately returns the stored response.
+  This post shows how to implement it with Apache APISIX.
+tags: [Community]
```
Comment (pottekkat) [inline blog/en/blog/2024/04/11/implement-idempotency-key-apisix.md]: Plugins or Ecosystem might be a better category option.

File: blog/en/blog/2024/04/11/implement-idempotency-key-apisix.md [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+image: https://static.apiseven.com/uploads/2024/04/09/0rfsRevo_stormtrooper-2899993.jpg
+---
+
+<head>
+    <link rel="canonical" href="https://blog.frankel.ch/implement-idempotency-key-apisix/" />
+</head>
+
+>Last week, I wrote an [analysis](https://blog.frankel.ch/fix-duplicate-api-requests/) of the [IETF Idempotency-Key specification](https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-idempotency-key-header-04). The specification aims to avoid duplicated requests. In short, the idea is for the client to send a unique key along with the request:
```
Comment (pottekkat) [inline blog/en/blog/2024/04/11/implement-idempotency-key-apisix.md]: Link to the article in the APISIX blog instead? Same for other links that are also available in the APISIX blog.

Comment (nfrankel) [reply blog/en/blog/2024/04/11/implement-idempotency-key-apisix.md]: I did it but I'm not sure it's a good idea. Cross domain links improve SEO and my domain is pretty good in that regard

**Review by pottekkat — APPROVED** [review]

File: blog/en/blog/2024/04/11/implement-idempotency-key-apisix.md (hunk 2 above)
Comment (pottekkat) [reply blog/en/blog/2024/04/11/implement-idempotency-key-apisix.md]: Internal links also help as well. But I'm also not sure what would be best.

Comment (nfrankel) [reply blog/en/blog/2024/04/11/implement-idempotency-key-apisix.md]: Me neither

**Review by moonming — APPROVED** [review]

**Pull request merged**